src/main/c/openssl/ossl_ssl.c

```diff
@@ -1441,10 +1441,9 @@ ossl_ssl_read_internal(int argc, VALUE *argv, VALUE self, int nonblock)
 	if (RSTRING_LEN(str) >= ilen)
 	    rb_str_modify(str);
 	else
-	    rb_str_modify_expand(str, ilen - RSTRING_LEN(str));
+	    rb_str_resize(str, ilen);
     }
     OBJ_TAINT(str);
-    rb_str_set_len(str, 0);
     if (ilen == 0)
```

Comment:
We need to comment this fix so I know to keep it when I import new OpenSSL changes for the next version. Why do we need this and MRI doesn't?

Thanks for the fix though!